meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockInfo.java

```diff
@@ -19,6 +19,231 @@
+      String ownerId = znRecord.getSimpleField(LockInfoAttribute.OWNER_ID.name());
+      String clientId = znRecord.getSimpleField(LockInfoAttribute.CLIENT_ID.name());
+      String clientData = znRecord.getSimpleField(LockInfoAttribute.CLIENT_DATA.name());
+      long grantTime = znRecord.getLongField(LockInfoAttribute.GRANTED_AT.name(), DEFAULT_GRANTED_AT_LONG);
+      long lastRenewalTime =
+          znRecord.getLongField(LockInfoAttribute.LAST_RENEWED_AT.name(), DEFAULT_LAST_RENEWED_AT_LONG);
+      long timeout = znRecord.getLongField(LockInfoAttribute.TIMEOUT.name(), DEFAULT_TIMEOUT_DURATION.toMillis());
+      Duration duration = Duration.of(timeout, ChronoUnit.MILLIS);
+      setLockInfoFields(lockId,ownerId, clientId, clientData, grantTime,
+          lastRenewalTime, duration);
+    }
+  }
+
+  /**
+   * Initialize a LockInfo with data for each field, set all null info fields to default data
+   * @param lockId value of LOCK_ID attribute
+   * @param ownerId value of OWNER_ID attribute
+   * @param clientId value of CLIENT_ID attribute
+   * @param clientData value of CLIENT_DATA attribute
+   * @param grantTime the time the lock was granted
+   * @param lastRenewalTime the last time the lock was renewed
+   * @param timeout value of TIMEOUT attribute
+   */
+  public LockInfo(String lockId, String ownerId, String clientId,
+                  String clientData, long grantTime, long lastRenewalTime, Duration timeout) {
+    this();
+    setLockInfoFields(lockId, ownerId, clientId, clientData, grantTime, lastRenewalTime, timeout);
+  }
+
+  /**
+   * Set each field of lock info to user provided values if the values
+   * are not null, null values are set to default values
+   * @param lockId value of LOCK_ID attribute
+   * @param ownerId value of OWNER_ID attribute
+   * @param clientId value of CLIENT_ID attribute
+   * @param clientData value of CLIENT_DATA attribute
+   * @param grantTime the time the lock was granted
+   * @param lastRenewalTime the last time the lock was renewed
+   * @param timeout value of TIMEOUT attribute
+   */
+  private void setLockInfoFields(String lockId, String ownerId, String clientId, String clientData, long grantTime, long lastRenewalTime,
+                                 Duration timeout) {
+    setLockId(lockId);
+    setOwnerId(ownerId);
+    setClientId(clientId);
+    setClientData(clientData);
+    setGrantedAt(grantTime);
+    setLastRenewedAt(lastRenewalTime);
+    setTimeout(timeout);
+  }
+
+  /**
+   * Set the value for LOCK_ID attribute of the lock
+   * @param lockId String representing the lock id
+   */
+  public void setLockId(String lockId) {
+    _record.setSimpleField(LockInfoAttribute.LOCK_ID.name(), lockId == null ? DEFAULT_LOCK_ID_TEXT : lockId);
+  }
+
+  /**
+   * Get the value for OWNER_ID attribute of the lock
+   * @param ownerId String representing the owner id
+   */
+  public void setOwnerId(String ownerId) {
```

Comment:
I would suggest adding more detail and some explanation on these fields.

Reply:
Hmm while I agree I am not sure what other information to put. The id of the owner is a unique string that represents the owner of each lock (similar to the rest of them). If you could give me an pointer to what additional information I can add that would be greatly appreciated, thank you!

Reply:
There are 3 `ID` fields in this info. We probably want to explain what's the difference, what info is suitable for each of these fields, maybe with an example.

Reply:
Okay that makes sense, I added more details though was hesitant on actual examples. Please let me know if this is detailed enough, thanks!